### PR TITLE
fix: update version constraints for TPG 5 compatibility

### PR DESCRIPTION
**Pull request by fveas**
The version must be set to < 6.0. Using < 5.0.0 as the constraint would lead to compatibility issues with version 5.1.0.

### CODE REVIEW

**Comment by fveas** [comment]
Hi @apeabody,

I've completed the changes and would appreciate if you could take some time to review my pull request. I believe your feedback will be invaluable, and I want to ensure everything aligns with our code standards before proceeding with the merge.

Please let me know if you have any initial thoughts or if there's anything specific you'd like me to address.

Thanks in advance for your time and expertise!

Best regards,
Felipe

File: versions.tf [hunk 1]
```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 4.0.0, < 5.0"
+      version = ">= 4.0.0, < 6.0"
```
Comment (apeabody) [inline versions.tf:22]: nit: let's use just `6` so in the future the [automated version widen](https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/pull/32) can be used 

```suggestion
      version = ">= 4.0.0, < 6"
```

File: versions.tf [hunk 2]
```diff
@@ -19,11 +19,11 @@ terraform {
...
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 4.0.0, < 5.0"
+      version = ">= 4.0.0, < 6.0"
```
Comment (apeabody) [inline versions.tf:26]: nit: let's use just `6` so in the future the [automated version widen](https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/pull/32) can be used 

```suggestion
      version = ">= 4.0.0, < 6"
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @fveas!

Can you also remove the `4.0` version constraint (all of line 21) in the example so the CI test won't be version constrained:
https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/examples/simple/versions.tf#L21

Let's also wider the `google` and `google-beta` version constraints in https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/test/setup/versions.tf#L22 from `~>` to `>=` so they can run with TPG 5 as well.

Thanks!

**Comment by apeabody** [comment]
Let's also use `fix:` for the PR subject so this will trigger a release and be included in the release notes.

**Comment by fveas** [comment]
> Let's also use `fix:` for the PR subject so this will trigger a release and be included in the release notes.

Thank you, @apeabody, for the detailed explanation and suggested changes. I've now applied them to the code. Please take a look when you have a moment.

**Comment by apeabody** [comment]
> Thanks for the contribution @fveas!
> 
> Can you also remove the `4.0` version constraint (all of line 21) in the example so the CI test won't be version constrained: https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/examples/simple/versions.tf#L21
> 
> Let's also wider the `google` and `google-beta` version constraints in https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/test/setup/versions.tf#L22 from `~>` to `>=` so they can run with TPG 5 as well.
> 
> Thanks!

Hi @fveas - Did you have an opportunity to make these changes as well?  Thanks!

**Comment by fveas** [comment]
> > Thanks for the contribution @fveas!
> > Can you also remove the `4.0` version constraint (all of line 21) in the example so the CI test won't be version constrained: https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/examples/simple/versions.tf#L21
> > Let's also wider the `google` and `google-beta` version constraints in https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/test/setup/versions.tf#L22 from `~>` to `>=` so they can run with TPG 5 as well.
> > Thanks!
> 
> Hi @fveas - Did you have an opportunity to make these changes as well? Thanks!

Hi @apeabody, I apologize for the oversight; I misunderstood the initial request. I've now made the requested changes. Can you please review them and let me know if everything looks good? Thanks for your patience and guidance.

**Comment by apeabody** [comment]
> Hi @apeabody, I apologize for the oversight; I misunderstood the initial request. I've now made the requested changes. For context, this is my first contribution on GitHub, so I'm still getting familiar with the process. Can you please review the changes and let me know if everything looks good? Thank you for your patience and guidance.

Hi @fveas - No worries at all. I'm only seeing a single file currently changed in this pull request, perhaps they didn't get included in you commit, etc?  You can see all the changes in the PR here: https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/pull/34/files Thanks!

File: versions.tf [hunk 3]
```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 4.0.0, < 5.0"
+      version = ">= 4.1"
```
Comment (apeabody) [inline versions.tf:22]:
```suggestion
      version = ">= 4.0.0, < 6"
```

File: versions.tf [hunk 4]
```diff
@@ -19,11 +19,11 @@ terraform {
...
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 4.0.0, < 5.0"
+      version = ">= 4.3"
```
Comment (apeabody) [inline versions.tf:26]:
```suggestion
      version = ">= 4.0.0, < 6"
```

File: versions.tf (hunk 3 above)
Comment (apeabody) [reply versions.tf:22]: `>= 4.1` was probably intended for https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/test/setup/versions.tf#L22

File: versions.tf (hunk 4 above)
Comment (apeabody) [reply versions.tf:26]: `>= 4.3` was probably intended for https://github.com/GoogleCloudPlatform/terraform-google-secret-manager/blob/main/test/setup/versions.tf#L26

**Comment by apeabody** [comment]
Breaking change to `secret_manager_secret` in https://github.com/hashicorp/terraform-provider-google/releases/tag/v5.0.0

**Comment by apeabody** [comment]
> Breaking change to `secret_manager_secret` in https://github.com/hashicorp/terraform-provider-google/releases/tag/v5.0.0
> 
> `secretmanager: removed automatic field in google_secret_manager_secret resource (https://github.com/hashicorp/terraform-provider-google/pull/15859)`

So, this change will actually need to be a breaking change `fix(TPG>=5)!:` and incorporate the above breaking change in the provider `secret_manager_secret` resource.

**Pull request closed without merging**